contracts/VetoERC20Voting.sol

```diff
@@ -0,0 +1,181 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC20Votes.sol";
+import "./interfaces/IVetoGuard.sol";
+import "./interfaces/IVetoERC20Voting.sol";
+import "./TransactionHasher.sol";
+import "@fractal-framework/core-contracts/contracts/ModuleBase.sol";
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+/// @notice A contract for casting veto votes with an ERC20 votes token
+contract VetoERC20Voting is IVetoERC20Voting, TransactionHasher, ModuleBase {
+    uint256 public vetoVotesThreshold;
+    IERC20Votes public votesToken;
+    IVetoGuard public vetoGuard;
+    mapping(bytes32 => uint256) public transactionVetoVotes;
+    mapping(address => mapping(bytes32 => bool)) public userHasVoted;
+
+  /// @notice Initializes the contract that can only be called once
+  /// @param _vetoVotesThreshold The number of votes required to veto a transaction
+  /// @param _votesToken The 
+  /// @param _vetoGuard The address of the VetoGuard contract
+  /// @param _accessControl The address of the access control contract
+    function initialize(
+        uint256 _vetoVotesThreshold,
+        address _votesToken,
+        address _vetoGuard,
+        address _accessControl
+    ) external initializer {
+        __initBase(_accessControl, msg.sender, "Veto ERC20 Voting");
+        vetoVotesThreshold = _vetoVotesThreshold;
+        votesToken = IERC20Votes(_votesToken);
+        vetoGuard = IVetoGuard(_vetoGuard);
+    }
+
+    /// @notice Allows the msg.sender to cast veto votes on the specified transaction
+    /// @param to Destination address.
+    /// @param value Ether value.
+    /// @param data Data payload.
+    /// @param operation Operation type.
+    /// @param safeTxGas Gas that should be used for the safe transaction.
+    /// @param baseGas Gas costs for that are independent of the transaction execution(e.g. base transaction fee, signature check, payment of the refund)
+    /// @param gasPrice Maximum gas price that should be used for this transaction.
+    /// @param gasToken Token address (or 0 if ETH) that is used for the payment.
+    /// @param refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+    function castVetoVote(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver
+    ) external {
+        // Get the transaction hash
+        bytes32 transactionHash = getTransactionHash(
```

Comment:
Yes we definitely could do it that way. Would save on gas for sure. 

Only reason I wrote it like this is to be consistent with the checkTransaction function - which has to take all these parameters to adhere to the interface.

Reply:
I see - that makes sense - Perhaps we leave for now for better readability - In the future - we would have a list of txs that are shown on the UI and a user will just choose one and it will autofill the data. Therefore - it may make sense just to have the hash which we can read on frontend and save a tad bit of gas

price of eth - no sell pressure right now.

Reply:
Oh yeah for sure, I like that

Reply:
Just pushed up a commit that:
- Makes castVetoVote within VetoERC20Voting take only the transactionHash as a param
- Makes getTransactionQueuedBlock within VetoGuard only take the transactionHash as a param

These versions of the function save gas for the user, since the hashing is done off chain. I didn't include a version of the function that takes all of the params, since it isn't necessary and uses more gas.